### PR TITLE
feat(memoryCache):support gets a list of values with fuzzy matching

### PR DESCRIPTION
**Pull request by 15168440402**

### CODE REVIEW

File: src/Caching/Masa.Utils.Caching.DistributedMemory/Masa.Utils.Caching.DistributedMemory.csproj [hunk 1]
```diff
@@ -10,6 +10,7 @@
     <PackageReference Include="Microsoft.Extensions.Caching.Memory" Version="$(MicrosoftPackageVersion)" />
     <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="$(MicrosoftPackageVersion)" />
     <PackageReference Include="System.Text.Json" Version="$(MicrosoftPackageVersion)" />
+    <PackageReference Include="Util.Reflection" Version="1.0.3" />
```
Comment (doddgu) [inline src/Caching/Masa.Utils.Caching.DistributedMemory/Masa.Utils.Caching.DistributedMemory.csproj:13]: I think we don't must depend on `Util.Reflection`.

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Caching/Masa.Utils.Caching.DistributedMemory/MemoryCacheClient.cs [hunk 2]
```diff
@@ -1,6 +1,9 @@
 // Copyright (c) MASA Stack All rights reserved.
 // Licensed under the MIT License. See LICENSE.txt in the project root for license information.
 
+using System.Collections;
```
Comment (zhenlei520) [inline src/Caching/Masa.Utils.Caching.DistributedMemory/MemoryCacheClient.cs:4]: global using

File: src/Caching/Masa.Utils.Caching.DistributedMemory/Interfaces/IMemoryCacheClient.cs [hunk 3]
```diff
@@ -23,4 +23,12 @@ public interface IMemoryCacheClient : ICacheClient
...
     /// <returns>The <see cref="Task"/> that represents the asynchronous opertion, containing the located value or null.</returns>
     Task<T?> GetAsync<T>(string key, Action<T?> valueChanged);
+
+    /// <summary>
+    /// Gets a list of values with fuzzy matching
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="pattern"></param>
+    /// <returns></returns>
+    Task<List<T>> FilterAsync<T>(string pattern);
```
Comment (zhenlei520) [inline src/Caching/Masa.Utils.Caching.DistributedMemory/Interfaces/IMemoryCacheClient.cs:33]: The name of the method for fuzzy matching memcached keys is not appropriate, and this should not be the responsibility of the distributed memcached, this method will ambiguous for those who use it

File: src/Caching/Masa.Utils.Caching.DistributedMemory/MemoryCacheClient.cs [hunk 4]
```diff
@@ -398,6 +409,21 @@ private void Set<T>(string key, T value, MemoryCacheEntryOptions? options = null
         }
     }
 
+    private List<string> GetCacheKeys()
+    {
+        const BindingFlags flags = BindingFlags.Instance | BindingFlags.NonPublic;
+        var entries = _cache.GetType().GetField("_entries", flags)!.GetValue(_cache);
```
Comment (zhenlei520) [inline src/Caching/Masa.Utils.Caching.DistributedMemory/MemoryCacheClient.cs:415]: performance can be optimized

**Pull request closed without merging**